r/PACKAGING.md

```diff
@@ -0,0 +1,128 @@
+- [ ] Pull request to modify the 
+  [autobrew script](https://github.com/autobrew/scripts/blob/master/apache-arrow)
+  to include any additions made to
+  [r/tools/autobrew](https://github.com/apache/arrow/blob/master/r/tools/autobrew).
+- [ ] Pull request to modify the
+  [PKGBUILD script](https://github.com/r-windows/rtools-packages/blob/master/mingw-w64-arrow/PKGBUILD)
+  to reflect changes in
+  [ci/PKGBUILD](https://github.com/apache/arrow/blob/master/ci/scripts/PKGBUILD),
+  uncommenting the line that says "uncomment to test the rc".
+
+Prepare and check the .tar.gz that will be released to CRAN.
+
+- [ ] `git fetch upstream && git checkout release-X.X.X-rcXX && git clean -f -d`
+- [ ] Unset the `ARROW_HOME` environment variable and then run `make build` (copies Arrow C++ into tools/cpp, prunes some unnecessary components, and runs `R CMD build`)
+- [ ] `devtools::check_built("arrow_X.X.X.tar.gz")` locally
+- [ ] Run reverse dependency checks
+
+Wait for the official release...
+  
+- [ ] Release vote passed!
+- [ ] Create a CRAN-release branch from the official release commit
+- [ ] Pick any commits that were made to master since the release commit that
+  were needed to fix CRAN-related submission issues (e.g., fixed URLs, fixes
+  to )
+- [ ] Remove badges from README.md
+- [ ] Run `urlchecker::url_check()` on the R directory
+- [ ] Bump the Version in DESCRIPTION to X.X.X
+- [ ] Create a PR entitled `WIP: [R] Verify CRAN release-10.0.1-rc0`. Add
+  a comment `@github-actions crossbow submit --group r` to run all R crossbow
+  jobs against the CRAN-specific release branch.
+- [ ] Regenerate arrow_X.X.X.tar.gz (i.e., `make build`)
+
+Update autobrew and r-windows PRs to use the *release* instead of the
+*release candidate*; ensure linux binary packages are available:
+
+- [ ] PR into autobrew/homebrew-core (apache-arrow autobrew formula)
+- [ ] PR into autobrew/homebrew-core (apache-arrow-static autobrew formula)
+- [ ] PR into autobrew/scripts
+- [ ] PR into r-windows/rtools-packages
+- [ ] Ensure linux binaries are available in the artifactory:
+  https://apache.jfrog.io/ui/repos/tree/General/arrow/r
+
+Check binary Arrow C++ distributions specific to the R package:
+
+- [ ] Upload the .tar.gz to [win-builder](https://win-builder.r-project.org/upload.aspx)
+  and confirm that the check is clean
+- [ ] Upload the .tar.gz to [MacBuilder](https://mac.r-project.org/macbuilder/submit.html)
+  and confirm that the check is clean
+- [ ] Check `install.packages("arrow_X.X.X.tar.gz")` on Ubuntu and ensure that the
+  hosted binaries are used
+- [ ] `devtools::check("arrow_X.X.X.tar.gz")` locally one more time (for luck)
+
+Submit!
+
+- [ ] Upload arrow_X.X.X.tar.gz to the
+  [CRAN submit page](https://xmpalantir.wu.ac.at/cransubmit/)
+- [ ] Confirm the submission email
+
+Wait for CRAN...
+
+- [ ] Accepted!
+- [ ] Tag the tip of the CRAN-specific release branch
```

Comment:
Tag how? This isn't something we've done before

Reply:
Hmm, I've created a branch on my fork called r-10.0.1, but we should probably do something more robust.

Reply:
@paleolimbot Any thoughts on this?